keylime/user_utils.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/python3
+
+'''
+SPDX-License-Identifier: Apache-2.0
+Copyright 2017 Massachusetts Institute of Technology.
+'''
+
+import os
+import grp
+import pwd
+
+from keylime import keylime_logging
+
+# Configure logger
+logger = keylime_logging.init_logging('privileges')
+
+
+def string_to_uidgid(userandgroup):
+    """ Translate the userandgroup string to uid and gid.
+        The userandgroup parameter must be a string of the format '<user>[:<group>]'.
+        User and group can be strings or integers. If no group is given, -1 will be
+        returned for gid.
+    """
+    params = userandgroup.split(':')
+    if len(params) > 2:
+        return None, None, "User and group '%s' are in wrong format. Expected <user>[:group]" % userandgroup
+
+    gid = -1
+    if len(params) == 2:
+        if params[1].isnumeric():
+            gid = int(params[1])
+        else:
+            try:
+                gr = grp.getgrnam(params[1])
+                gid = gr.gr_gid
+            except Exception as e:
+                return None, None, 'Could not resolve group %s: %s' % (params[1], str(e))
+
+    if params[0].isnumeric():
+        uid = int(params[0])
+    else:
+        try:
+            passwd = pwd.getpwnam(params[0])
+            uid = passwd.pw_uid
+        except Exception as e:
+            return None, None, 'Could not resolve user %s: %s' % (params[0], str(e))
+
+    return uid, gid, None
```

Comment:
Hm ... So, yes, it's a go pattern.  You mean it should throw a RuntimeError()? What I don't like about python is that sometimes you don't know what all exceptions can surface from functions. I would at least not kill the process from within this library function here but let the caller handle the error...

Reply:
Yeah it is not ideal how exceptions in Python work. I would raise in this case a ValueError because it could not resolve the user/group. We can start to add comments for the functions with something like `:raises ValueError: if user or group could not be resolved`, so that we can more easily see which of our functions raise exceptions.